### DISCUSSION TITLE
configure a config client in a container

### DISCUSSION THREAD

**Discussion by bjornTennander** [hyness/spring-cloud-config-server#45] [Q&A] [answered]
Hi!
Sorry to use this discussion forum of you project if not appropriate. Please direct me to a proper channel.

I'm evaluating your config server for our project. I've managed to configure a client running on the host, but failing when trying to configure a spring client within the same container. I'm getting the below message at the start of the client.
I'm using Docker desktop, and using a shell within a container I get an expected answer from curl http://configService:888/client-name/dev. I'm setting the environment variable spring.config.import=http://configService:8888/ in the docker-compose file.

The error message is as follows. What am I doing wrong?
/Björn

Description:
Config data resource 'URL [http://configService:8888/]' via location 'http://configService:8888/' does not exist

**Comment by bjornTennander** [comment]
With an application.yml file for the clients, having 
spring:
  config:
    import: "optional:configserver:"
  cloud:
    config:
      uri: ${config.uri:http://localhost:8888/}

I could set the environment variable config.uri in the docker-compose file to point to configService:8888. And it worked.
/Björn

**Comment by hyness** [comment]
Hello Bjorn,

You can’t have periods in environment variables. Spring Boot’s flexible naming allows for you to replace the periods with underscores. So, try specifying it as SPRING_CONFIG_IMPORT

**Reply by bjornTennander** [reply]
Thanks, I've read that, I remember now, but it seems to work under some situations. I'll keep this in mind when moving further into a container setup and deployment in the cloud, cause currently I got it to work for our development environment.

**Comment by hyness** [comment]
Just a friendly reminder, you are not required to use environment variables.  You can mount an `application.yml` in your container.  The `README.md` only shows docker examples, but there are also a [bunch of example docker-compose files](https://github.com/hyness/spring-cloud-config-server/tree/master/examples).  Each example shows how to configure the server with a file, system properties and environment variables.  These examples will be updated for Spring Boot 2.4 in the near future, but all of them work.

**Reply by bjornTennander** [reply, marked as answer]
Thanks,
My problem was the clients, and the decision for our services to use the same application.yml file for booting regardless of environment. But currently I have no longer problems.
I put some ${variable:default-value} in the application.yml file making it possible to change value by adding environment variables in the compose file. And it works. This happens before Spring I guess. There is so many ways to do things in Spring. Hard for a beginner to understand the best way.
Interesting to look into your implementation, and understand that you really need to learn how to control the behavior of a Spring application by all its annotations.
